zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/IdentitySetupInitializeProcessor.java

```diff
@@ -41,108 +36,63 @@ public final class IdentitySetupInitializeProcessor
   private final KeyGenerator keyGenerator;
   private final TypedCommandWriter commandWriter;
   private final StateWriter stateWriter;
-  private final RoleState roleState;
 
-  public IdentitySetupInitializeProcessor(
-      final Writers writers,
-      final KeyGenerator keyGenerator,
-      final ProcessingState processingState) {
+  public IdentitySetupInitializeProcessor(final Writers writers, final KeyGenerator keyGenerator) {
     this.keyGenerator = keyGenerator;
     commandWriter = writers.command();
     stateWriter = writers.state();
-    roleState = processingState.getRoleState();
   }
 
   @Override
   public void processRecord(final TypedRecord<IdentitySetupRecord> command) {
     final var initializationKey = keyGenerator.nextKey();
     final var setupRecord = command.getValue();
 
-    final var defaultRole = setupRecord.getDefaultRole();
-    final var roleExists = roleState.getRole(defaultRole.getRoleId()).isPresent();
-    createDefaultRole(initializationKey, defaultRole);
-
-    final var defaultTenant = setupRecord.getDefaultTenant();
-    createDefaultTenant(initializationKey, defaultTenant);
-
-    final var users = setupRecord.getUsers();
-    createUsers(initializationKey, users, defaultRole, defaultTenant);
-
-    final var mappings = setupRecord.getMappings();
-    createMappings(initializationKey, mappings, defaultRole, defaultTenant);
-
-    if (!roleExists) {
-      addAllPermissions(initializationKey, defaultRole);
-    }
+    createRoles(initializationKey, setupRecord.getRoles());
+    createDefaultTenant(initializationKey, setupRecord.getDefaultTenant());
```

Comment:
❓ Perhaps out of scope for this PR but I'm wondering if we need to restrict this to the default tenant, could we just have it as `createTenants` and always add a "default" tenant to the list provided? This would then support the future of creating tenants in config

Reply:
It's out of scope but if we ever want to have config for initial tenants, the required changes are trivial.